offix-offline/src/main/java/org/aerogear/offixoffline/NetworkChangeReceiver.kt

```diff
@@ -0,0 +1,42 @@
+package org.aerogear.offixoffline
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+
+/*
+ BroadcastReceiver that listens for connectivity change while the app is in foreground (in-memory).
+ In it's onReceive() method we check the network connection. If the user is connected to the net, then get access to the arraylist
+ of mutations and callbacks, get the apollo client provided by the user and make a call to the server.
+ */
+class NetworkChangeReceiver : BroadcastReceiver() {
+
+    val ctx = Offline.getContextOffline()
+    val TAG = javaClass.simpleName
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        val isNetworkAvail = Offline.isNetwork()
+        Log.d(TAG, "${intent?.data}")
+        if (intent?.action == "android.net.conn.CONNECTIVITY_CHANGE") {
+            if (isNetworkAvail) {
+                Log.d(TAG, "Network connectivity change $isNetworkAvail")
+                val mutationList = Utils.offlineArrayList
```

Comment:
`Utils.offlineArrayList` using static globals to hold the data. 
I think we might simply use OfflineClient object to hold that inside the application.
This approach may lead co couple bugs

Reply:
Done, removed the Utils class 👍